src/generator/type_inference_visitor.ts

```diff
@@ -157,26 +155,19 @@ class TypeInferenceVisitor extends Visitor {
         super.visit(node.right);
 
         if (isIdentifier(node.left)) {
-            const rightSideType = this.expressionTypes.get(node.right);
+            let rightSideType = this.getTypeOfExpression(node.right);
 
-            if (rightSideType === undefined) {
-                throw new Error(`Unknown type for right side of assignment to ${node.left.name}`);
+            if (node.operator !== '=') {
+                rightSideType = getCommonNumberType(this.getTypeOfIdentifier(node.left), rightSideType);
```

Comment:
Wouldn't this change the type of the right side?

Reply:
No the type of the right side is not changed during an assignment.

Reply:
The discussed optimization of setting the type of the right node doesn't work, because array member access expressions are allowed to be on the right side of a shorthand assignment. Their types must be correct, in order to call the `load` method on the correct operations instance in the generator visitor.

Reply:
`getArrayElement` of the `GeneratorVisitor` doesn't distinguish between int and double at the moment so this shouldn't be a problem.